include/inexor/vulkan-renderer/imgui.hpp

```diff
@@ -48,7 +48,7 @@ class ImGUIOverlay {
     /// @param swapchain A reference to the swapchain
     /// @param render_graph A pointer to the render graph
     /// @param back_buffer A pointer to the target of the ImGUI rendering
-    ImGUIOverlay(const wrapper::Device &device, const wrapper::Swapchain &swapchain, RenderGraph *render_graph,
+    ImGUIOverlay(wrapper::Device &device, const wrapper::Swapchain &swapchain, RenderGraph *render_graph,
```

Comment:
`wrapper::Device &device`

 is that safe?

Reply:
You mean not taking it as a const reference? Yes that is safe and we must do this, because we store the reference as a non-const reference because we need to call some methods inside of the device wrapper which are not const. This would not be possible if we stored a `const wrapper::Device &m_device;`.

Reply:
Btw we do that in some other code parts too after this pull request, for example in the `RenderGraph` constructor or `GpuTexture` constructor.

Reply:
Oh wait.. maybe we can still take a const reference and mark all the used methods inside of the `Device` wrapper as const. Then we make the members of Device wrapper `mutable`?

Reply:
It really works! Nice!

Reply:
I will resolve this conversation now. I think yeetari will say something about that as well in his review.